src/clr-angular/forms/select/providers/option-selection.service.spec.ts

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2016-2018 VMware, Inc. All Rights Reserved.
+ * This software is released under MIT license.
+ * The full license information can be found in LICENSE in the root directory of this project.
+ */
+
+import {ElementRef} from "@angular/core";
+import {Subscription} from "rxjs";
+
+import {IfOpenService} from "../../../utils/conditional/if-open.service";
+import {ClrOption} from "../option";
+
+import {OptionSelectionService} from "./option-selection.service";
+
+export default function() {
+    describe("Option Selection Service", () => {
+        let optionSelectionService: OptionSelectionService;
+        let fakeOption1: ClrOption;
+        let fakeOption2: ClrOption;
+
+        beforeEach(() => {
+            optionSelectionService = new OptionSelectionService();
+            fakeOption1 = new ClrOption(new IfOpenService(), new ElementRef(null), null, optionSelectionService);
+            fakeOption2 = new ClrOption(new IfOpenService(), new ElementRef(null), null, optionSelectionService);
+        });
+
+        it("updates the selection with the selected option", () => {
+            let selectedOption: ClrOption;
+            const subscription: Subscription =
+                optionSelectionService.selectionChanged.subscribe((option: ClrOption) => {
+                    selectedOption = option;
+                });
+
+            optionSelectionService.updateSelection(fakeOption1);
+
+            expect(selectedOption).toBe(fakeOption1);
+
+            subscription.unsubscribe();
+        });
+
+        it("replaces the current selection with the new selection", () => {
+            let selectedOption: ClrOption;
+            const subscription: Subscription =
+                optionSelectionService.selectionChanged.subscribe((option: ClrOption) => {
+                    selectedOption = option;
+                });
+
+            optionSelectionService.updateSelection(fakeOption1);
+
+            expect(selectedOption).toBe(fakeOption1);
+
+            optionSelectionService.updateSelection(fakeOption2);
+
+            expect(selectedOption).toBe(fakeOption2);
+
+            subscription.unsubscribe();
+        });
+
+        it("updates the selected flag of the option when it is selected", () => {
+            expect(fakeOption1.selected).toBe(false);
```

Comment:
When you move to a model-based selection, the options will need to listen to selection changes to update their own flag.

Reply:
done